frontend/src/features/entity_selector/EntitySelector.tsx

```diff
@@ -184,21 +202,82 @@ const EntitySelectorInnerAuth = ({
     setLoading(false);
   }, [onChange, options?.comparisonsCanBePublic, pollName, uid]);
 
+  /**
+   * Load the user's rating.
+   *
+   * Do not load anything if the entity availability is unknown, or if an
+   * entity auto-reload has been asked.
+   */
   useEffect(() => {
-    if (isUidValid(uid) && rating == null) {
-      loadRating();
+    if (entityAvailability === ENTITY_AVAILABILITY.AVAILABLE || !autoReload) {
+      if (onEntityAvailable) {
+        onEntityAvailable();
+      }
+    } else if (entityAvailability === ENTITY_AVAILABILITY.UNAVAILABLE) {
+      if (onEntityUnavailable) {
+        onEntityUnavailable();
+      }
+    }
+
+    if (
+      entityAvailability === ENTITY_AVAILABILITY.AVAILABLE ||
+      (entityAvailability === ENTITY_AVAILABILITY.UNAVAILABLE && !autoReload)
+    ) {
+      if (isUidValid(uid) && rating == null) {
+        loadRating();
+      }
+    }
+  }, [
+    autoReload,
+    entityAvailability,
+    loadRating,
+    onEntityAvailable,
+    onEntityUnavailable,
+    rating,
+    uid,
+  ]);
+
+  /**
+   * Ask a new entity if an auto-reload has been asked.
+   */
+  useEffect(() => {
+    if (entityAvailability === ENTITY_AVAILABILITY.UNAVAILABLE) {
+      if (autoReload && !loading) {
+        setLoading(true);
+        getVideoForComparison(
```

Comment:
As pointed by @GresilleSiffle, this component is not specific to "videos", so we should check the current poll before calling `getVideoForComparison`.